### PR TITLE
Implemented Pseudonymize action

### LINKED ISSUE #41: Use hash function (+salt) to pseudonymize columns
The de-identifier should use a hash function to generate pseudonyms for tags with the `U` action (from the table [here](https://dicom.nema.org/medical/dicom/current/output/html/part15.html#table_E.1-1a)).

### PR DESCRIPTION
**Pull request by marijncv**
Fixes #41

### CODE REVIEW

File: src/main/scala/spark/dicom/deidentifier/DicomDeidentifier.scala [hunk 1]
```diff
@@ -39,11 +39,14 @@ object DicomDeidentifier {
...
     *   configuration settings to be used in de-identification
+    * @param salt
+    *   salt to use in the pseudonumization action
     */
   def getAction(
       deidElem: DicomDeidElem,
       vr: VR,
-      config: Map[DeidOption, Boolean]
+      config: Map[DeidOption, Boolean] = Map.empty,
+      salt: String = scala.util.Random.alphanumeric take 20 mkString
```
Comment (GuillaumeDesforges) [inline src/main/scala/spark/dicom/deidentifier/DicomDeidentifier.scala]: It's better not to use effectful code as default function arguments value.

File: src/main/scala/spark/dicom/deidentifier/DicomDeidentifier.scala [hunk 2]
```diff
@@ -72,10 +75,13 @@ object DicomDeidentifier {
...
     * @param config
     *   configuration settings to be used in de-identification
+    * @param salt
+    *   salt to use in the pseudonumization action
     */
   def deidentify(
       dataframe: DataFrame,
-      config: Map[DeidOption, Boolean] = Map.empty
+      config: Map[DeidOption, Boolean] = Map.empty,
+      salt: String = scala.util.Random.alphanumeric take 20 mkString
```
Comment (GuillaumeDesforges) [inline src/main/scala/spark/dicom/deidentifier/DicomDeidentifier.scala]: It's better not to use effectful code as default function arguments value.

**Review by GuillaumeDesforges — APPROVED** [review]

File: src/main/scala/spark/dicom/deidentifier/DicomDeidentifier.scala (hunk 1 above)
Comment (marijncv) [reply src/main/scala/spark/dicom/deidentifier/DicomDeidentifier.scala]: Makes sense. Would you propose no default value? Or just empty string as default

Comment (GuillaumeDesforges) [reply src/main/scala/spark/dicom/deidentifier/DicomDeidentifier.scala]: In the lowest levels of the API, I prefer using as little default values as possible, but that a personal preference.

**Pull request merged**